File: src/main/java/com/hobojoe/enderore/item/ModItems.java
package com.hobojoe.enderore.item;

import net.minecraft.creativetab.CreativeTabs;
import net.minecraft.item.Item;
import net.minecraftforge.fml.common.registry.GameRegistry;

/**
 * Created by Joseph on 11/18/2016.
 */
public class ModItems {

    public static ItemBase dustEnder;

    public static void init() {
        dustEnder = register(new ItemBase("dustEnder").setCreativeTab(CreativeTabs.MATERIALS));
    }

    private static <T extends Item> T register(T item) {
        GameRegistry.register(item);

        if (item instanceof ItemBase) {
            ((ItemBase)item).registerItemModel();
        }

        return item;
    }
}

File: src/main/java/com/hobojoe/enderore/item/ModItems.kt
package com.hobojoe.enderore.item

import com.hobojoe.enderore.EnderOre
import net.minecraftforge.fml.common.registry.GameRegistry
import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly

/**
 * Created by Joseph on 11/18/2016.
 */
object ModItems {

    const val id = EnderOre.MODID

    @GameRegistry.ObjectHolder("$id:dust_ender")
    lateinit var dustEnder: ItemEnderDust

    @SideOnly(Side.CLIENT)
    fun initModels() {
        dustEnder.initModel()
        dustEnder.initOreDict()
    }
}